src/dispatch/plugins/dispatch_core/plugin.py

```diff
@@ -273,6 +279,102 @@ def get(
         return recommendation.matches
 
 
+from dispatch.database.core import get_session, get_organization_session
+
+
+class DispatchMfaPlugin(MultiFactorAuthenticationPlugin):
+    title = "Dispatch Plugin - Multi Factor Authentication"
+    slug = "dispatch-auth-mfa"
+    description = "Uses dispatch itself to validate external requests."
+    version = dispatch_plugin.__version__
+
+    author = "Netflix"
+    author_url = "https://github.com/netflix/dispatch.git"
+
+    def wait_for_challenge(self, challenge_id: UUID, db_session: Session, timeout: int = 300):
+        """Waits for a multi-factor authentication challenge."""
+        import time
+
+        start_time = time.time()
+
+        while time.time() - start_time < timeout:
+            with get_organization_session(organization_slug="default") as session:
```

Comment:
If this is within a web request shouldn't it already be scoped to the organization?